Repository/Resource/PhpcrResource.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace Symfony\Cmf\Component\Resource\Repository\Resource;
+
+use Puli\Repository\Api\Resource\Resource;
+use Puli\Repository\Api\ResourceRepository;
+use PHPCR\NodeInterface;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use Symfony\Cmf\Component\Resource\Repository\Resource\Metadata\PhpcrMetadata;
+
+/**
+ * Resource representing a PHPCR node
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class PhpcrResource implements Resource
+{
+    private $node;
+    private $path;
+    private $repoPath;
+
+    /**
+     * @param string $path
+     * @param NodeInterface $node
+     */
+    public function __construct($path, NodeInterface $node)
+    {
+        $this->node = $node;
+        $this->repoPath = $path;
+        $this->path = $path;
+    }
+
+    /**
+     * Return the PHPCR node which this resource
+     * represents.
+     *
+     * @return NodeInterface
+     */
+    public function getNode()
+    {
+        return $this->node;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getPath()
+    {
+        return $this->path;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getName()
+    {
+        return $this->node->getName();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getChild($relPath)
+    {
+        $child = $this->node->getNode($relPath);
+        return new self($this->path . '/' . $relPath, $child);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function hasChild($relPath)
+    {
+        return $this->node->hasNode($relPath);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function hasChildren()
+    {
+        return $this->node->hasNodes();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function listChildren()
+    {
+        $collection = new ArrayResourceCollection();
+        foreach ($this->node->getNodes() as $node) {
+            $collection->add(new self($this->path . '/' . $node->getName(), $node));
+        }
+
+        return $collection;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getMetadata()
+    {
+        return new PhpcrMetadata($this->node);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getRepository()
+    {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getRepositoryPath()
+    {
+        return $this->repoPath;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function attachTo(ResourceRepository $repo, $path = null)
```

Comment:
Is the following workflow what you would expect?

````php
$apcRepository = new ApcRepository();

$phpcrResource = $phpcrRepository->get('some/resource');
$apcRepository->add('/foo', $phpcrResource);

$apcResource = $apcRepository->get('/foo');
````

If so:

- How would I access the properties (f.e. title, body, tags, language, photo) of the
  PHPCR resource from the new `ApcResource`?

Would I need to implement a specialized repository which was aware of `PhpcrResource`?

(p.s. we could probably implement something like a `MappedPropertyResource` interface or similar
which would allow for consitent access to database records).